### PR TITLE
Feat/pancake zap multi token

### PR DESCRIPTION
**Pull request by tienkane**

### CODE REVIEW

File: packages/hooks/src/use-device-screen.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from "react";
+
+export function useDeviceScreen(): {
```
Comment (viet-nv) [inline packages/hooks/src/use-device-screen.ts:3]: I think we can change this to a generic hook like `useWindowSize` 
isMobile and isTable can be different depend on each usecase. just return with and height of the screen, the consumer will decide how to use it.

File: packages/ui/tailwind.config.ts [hunk 2]
```diff
@@ -3,8 +3,35 @@ import type { Config } from "tailwindcss";
 import sharedConfig from "@kyber/tailwind-config";
 
 const config: Pick<Config, "presets" | "content"> = {
-  content: ["./src/**/*.tsx"],
+    darkMode: ["class"],
+    content: ["./src/**/*.tsx"],
   presets: [sharedConfig],
+    theme: {
+    	extend: {
```
Comment (viet-nv) [inline packages/ui/tailwind.config.ts:10]: Please re-format this file

File: packages/pancake-liquidity-widgets/src/components/Widget/Widget.scss [hunk 3]
```diff
@@ -16,16 +17,17 @@
...
   border-radius: 24px;
   background: var(--ks-lw-cardBackground);
   color: var(--ks-lw-textPrimary);
-  // max-width: 680px;
   font-family: Kanit, Sans-serif;
   position: relative;
   transition: all 0.2s;
   box-shadow: 0px 1px 2px 0px #00000014;
   border: 1px solid var(--ks-lw-cardBorder);
+  max-width: 960px;
```
Comment (viet-nv) [inline packages/pancake-liquidity-widgets/src/components/Widget/Widget.scss:32]: dont use max-width here. let partner do it by themselves

File: packages/pancake-liquidity-widgets/src/components/Widget/index.tsx [hunk 4]
```diff
@@ -1,67 +1,71 @@
...
   poolAddress: string;
   positionId?: string;
-  onDismiss: () => void;
-  onTxSubmit?: (txHash: string) => void;
   feeAddress?: string;
   feePcm?: number;
   source: string;
   includedSources?: string;
   excludedSources?: string;
+  initDepositTokens: string;
```
Comment (viet-nv) [inline packages/pancake-liquidity-widgets/src/components/Widget/index.tsx:34]: Help me to add these new params into readme

File: packages/pancake-liquidity-widgets/src/constants/index.ts [hunk 5]
```diff
@@ -255,4 +255,45 @@ export const PANCAKE_NFT_MANAGER_CONTRACT: { [chainId: number]: Address } = {
...
+  8453: "0xcA11bde05977b3631167028862bE2a173976CA11",
+  81457: "0xcA11bde05977b3631167028862bE2a173976CA11",
+  5000: "0xcA11bde05977b3631167028862bE2a173976CA11",
+};
+
 export const BASE_BPS = 10_000;
+
+export const PATHS = {
+  KYBERSWAP_SETTING_API: "https://ks-setting.kyberswap.com/api/v1/tokens",
+  KYBERSWAP_PRICE_API: "https://price.kyberswap.com",
```
Comment (viet-nv) [inline packages/pancake-liquidity-widgets/src/constants/index.ts:280]: just wait for me on this a bit, let take this chance to remove old price service then we can deprecate it soon

File: packages/pancake-liquidity-widgets/src/hooks/useMarketPrice.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            ) => {
+              tokensInClone[index].price = item.marketPrice || item.price || 0;
+            }
+          );
+          setTokensIn(tokensInClone);
+        });
+    };
+
+    getPrices();
+    const i = setInterval(() => getPrices, 5 * 60_000);
```
Comment (viet-nv) [inline packages/pancake-liquidity-widgets/src/hooks/useMarketPrice.ts:60]: why 5 mins?

File: packages/pancake-liquidity-widgets/src/hooks/useMarketPrice.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+} from "@/constants";
+import { useWeb3Provider } from "@/hooks/useProvider";
+import { PancakeTokenAdvanced } from "@/types/zapInTypes";
+
+export default function useMarketPrice({
+  tokensIn,
+  setTokensIn,
+}: {
+  tokensIn: PancakeTokenAdvanced[];
+  setTokensIn: (value: PancakeTokenAdvanced[]) => void;
```
Comment (viet-nv) [inline packages/pancake-liquidity-widgets/src/hooks/useMarketPrice.ts:16]: this confuse me, why a hook named `useMarketPrice` can `setTokensIn`?
If I write this hook:
- input: tokens (or tokenAddresses) 
- output: 
    + loading 
    + prices 
    + refetch
    
  Then if you want to enrich token data, the logic should be in the component where consuming this hook

Comment (viet-nv) [reply packages/pancake-liquidity-widgets/src/hooks/useMarketPrice.ts:16]: or if you want to enrich price info here, I suggest:
- create a separate hook to get prices only: useMarketPrice hook
- rename this to: useEnrichPriceInfo

File: packages/pancake-liquidity-widgets/src/hooks/useTokenBalance.ts [hunk 8]
```diff
@@ -1,61 +1,79 @@
...
+import { useWeb3Provider } from "@/hooks/useProvider";
+import { NATIVE_TOKEN_ADDRESS, NetworkInfo } from "@/constants";
 
-import { useWeb3Provider } from "./useProvider";
+export default function useTokenBalance({
+  tokensIn,
+  setTokensIn,
+}: {
+  tokensIn: PancakeTokenAdvanced[];
+  setTokensIn: (value: PancakeTokenAdvanced[]) => void;
```
Comment (viet-nv) [inline packages/pancake-liquidity-widgets/src/hooks/useTokenBalance.ts:12]: same with useMarketPrice hook.

File: packages/pancake-liquidity-widgets/src/hooks/useTokens.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        allTokens,
+        getToken,
+      }}
+    >
+      {children}
+    </TokenListContext.Provider>
+  );
+};
+
+export const useTokens = () => {
```
Comment (viet-nv) [inline packages/pancake-liquidity-widgets/src/hooks/useTokens.tsx:98]: do we use this anymore?

Comment (tienkane) [reply packages/pancake-liquidity-widgets/src/hooks/useTokens.tsx:98]: Yes

**Review by viet-nv — APPROVED** [review]

**Pull request merged**